### PR TITLE
Lanierd/hydra 746

### PR DESCRIPTION
**Pull request by lanierd-adsk**

### CODE REVIEW

**Comment by lanierd-adsk** [comment]
FYI : @ppt-adsk @roopavr-adsk @vlasovi @debloip-adsk

File: lib/mayaHydra/flowViewportAPIExamples/flowViewportAPILocator/mhFlowViewportAPILocator.cpp [hunk 1]
```diff
@@ -437,81 +384,80 @@ void FlowViewportAPIMayaLocator::SetupFlowViewportInterfaces()
...
-        MDagPath parentDagPath = MDagPath::getAPathTo(currentMObj, &stat);
-        CHECK_MSTATUS(stat);
-        parentDagPath.pop();
-        MObject parentObj = parentDagPath.node();
-        _cbParentAttributeChangedId = MNodeMessage::addAttributeChangedCallback(parentObj, transformAttributeChangedCallback, this);
-        
-        _init                               = true;
+    //The MObject can change if the node gets deleted and deletion being undone
+    if (! _thisMObject.isValid()){
+        setupFlowViewportInterfaces();
```
Comment (ppt-adsk) [inline lib/mayaHydra/flowViewportAPIExamples/flowViewportAPILocator/mhFlowViewportAPILocator.cpp]: Perhaps not in this pull request, but can we move this to the postConstructor() at some point?

Comment (lanierd-adsk) [reply lib/mayaHydra/flowViewportAPIExamples/flowViewportAPILocator/mhFlowViewportAPILocator.cpp:391]: Yes I am doing it in this PR : https://github.com/Autodesk/maya-hydra/pull/85

Comment (lanierd-adsk) [reply lib/mayaHydra/flowViewportAPIExamples/flowViewportAPILocator/mhFlowViewportAPILocator.cpp]: No I cannot, in the postConstructor, there is a valid MObject but it's going to change later and it's not the one I should use. So no, there is no possibility to move this to the postConstructor.

**Review by ppt-adsk — DISMISSED** [review]

File: test/lib/mayaUsd/render/mayaToHydra/testFlowViewportAPI.py
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testFlowViewportAPI.py:1]: Don't forget to move the file new at the end of each test method to a single tearDown().

**Review by ppt-adsk — COMMENTED** [review]
Looks good to me, only missing the change for tearDown().

**Review by ppt-adsk — APPROVED** [review]

**Comment by lanierd-adsk** [comment]
Closing as it needs a rebase to fix merge conflicts. I will open a new PR for this change.

**Pull request closed without merging**